### PR TITLE
Budget-app

### PR DESCRIPTION
**Pull request by JavierAybar**
**_In this project I did:_**

- Splash screen 
  ✅  A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  ✅  The user should be able to register in the app with full name, email and password (all mandatory).
  ✅ The user can log into the app using email and password.
  ✅  If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (categories page)
  ✅ When the user logs in, they are presented with the categories page.
  ✅ For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  ✅ When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  ✅ There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  ✅ For a given category, the list of transactions is presented, ordered by the most recent.
  ✅ At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  ✅ There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  ✅ When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  ✅ The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  ✅  The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  ✅  When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    ✅  name (mandatory)
    ✅  amount (mandatory)
    ✅ categories (mandatory at least one)
  ✅  The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
  ✅  When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

#### Testing requirements
✅  Create unit and integration tests for all the most important components of your RoR application.

#### Technical requirements

✅  You should use Postgres as your database.
✅  You should use devise for authentication.
✅  You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
✅  You can use a view template engine of your choice (.erb, .slim, .haml).
✅  The project should be deployed and accessible online.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,189 @@
...
+- Add all website files in the new repository
+- Go to Settings/Pages
+- Go to branch and change from 'none' to 'main'
+- Click Save
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## Walkthrough video 🎥<a name="#walkthrough-video"></a>
+
+[Link](https://www.loom.com/share/fa3a379f05664d22975321f84fae4c53?sid=cb62457a-c976-48d7-b31e-ad4dbde345d9)
```
Comment (omar-muhamad) [inline README.md]: - [x] In the presentation video you provided, you didn't highlight some interesting piece of code or something you built that you are very proud of. Please add that to you video as per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=you%20should%20also%20highlight%20some%20interesting%20piece%20of%20code%20or%20something%20you%20built%20that%20you%20are%20very%20proud%20of.).

File: app/views/groups/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  <h2 class="text-xl font-semibold">Categories</h2>
+    <i class="text-lg fa-solid fa-magnifying-glass"></i>
+</nav>
+ 
+<div class="w-full h-[595px] max-w-md p-5 bg-zinc-50 border-gray-200 rounded-lg shadow dark:bg-white">
+   <div class="flow-root">
+        <ul role="list" class="divide-y divide-gray-200 flex flex-col gap-3">
+ <% @groups.each do |group| %>
+ <a class="flex items-center bg-white border border-gray-200 rounded-sm shadow  h-[80px] " href="<%= group_path(group) %>">
+            <li class="py-3 sm:py-4">
```
Comment (omar-muhamad) [inline app/views/groups/index.html.erb]: - [x] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=When%20the%20user%20clicks%20(or%20taps)%20on%20a%20category%20item%2C%20the%20application%20navigates%20to%20the%20transactions%20page%20for%20that%20category.), When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category, not for the category show page as you did. 
For your naming scheme, this link should redirect to `group_operations_path` which is `operations#index` page, not to `group_path` which is `groups#show` page as you did. Please fix that.

File: app/views/operations/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+
+<h1>Operations</h1>
+
```
Comment (omar-muhamad) [inline app/views/operations/index.html.erb]: - [x] As I mentioned above, This page (`operations#index` page) is supposed to be your transaction page. 
As per these [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=a%20new%20category.-,Transactions%20page,-For%20a%20given), Please edit this page to serve the following:
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
When the user clicks on the "Back" button (<), the user navigates to the home page.

File: spec/models/groups_spec.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+RSpec.describe Group, type: :model do
+  it 'is valid with valid attributes' do
+    user = User.create(name: 'Javier', email: 'email@gmail.com', password: 'password')
+    group = Group.new(user:, name: 'Example', icon: 'example.png')
+    expect(group).to be_valid
+  end
```
Comment (omar-muhamad) [inline spec/models/groups_spec.rb:8]: - [x] Good job adding model and controller specs, As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=integration%20tests%20for%20all%20the%20most%20important%20components%20of%20your%20RoR%20application.), Please also add integration(features) specs for used app `views`.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @JavierAybar,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by gemmen29 — APPROVED** [review]
Hi @JavierAybar, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**